src/helpers/stateful_list.rs

```diff
@@ -77,4 +78,26 @@ impl<T> StatefulList<T> {
     pub fn unselect(&mut self) {
         self.state.select(None);
     }
+    pub fn select(&mut self, i: usize) {
+        self.curr = i;
+        self.state.select(Some(i));
+    }
+}
+
+impl<T: ToString> StatefulList<T> {
+    pub fn find_by_path(&self, s: &PathBuf) -> usize {
+        let mut i = 0;
+
+        for n in 0 .. self.items.len() {
+            if s.ends_with(self.items[n].to_string()) {
+                i = n;
+                break;
+            }
+        }
+
+        i
+    }
+    pub fn select_by_path(&mut self, s: &PathBuf) {
```

Comment:
Would it make more sense to directly use `String` here, rather than `PathBuf`?

Reply:
Hey, thanks for this PR. I will do testing this weekend to see if this works. I remeber trying to do something similar and thought I ran into issues, but if your saying it works on your machine I'll try testing this code on mine.